### PR TITLE
Auto-update preview on pattern change (fixes #27)

### LINKED ISSUE #27: WIP: Update preview when pattern code changes
... if the component has e.g. a new styling

### PR DESCRIPTION
**Pull request by TheReincarnator**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@ designkit
 node_modules
 package-lock.json
 yarn.lock
+yarn-error.lock
```
Comment (Jumace) [inline .gitignore]: This is PR contains a really nice feature, looking forward to use it! But I need to be a bit picky, how is the 'yarn-error.lock' file connected to this feature?
And when did we switch to yarn? :D

Comment (TheReincarnator) [reply .gitignore]: @Jumace We didn't officially. Currently, some use npm, some yarn. I even use both :-D

Comment (TheReincarnator) [reply .gitignore]: done.

File: src/styleguide/analyzer/styleguide-analyzer.ts [hunk 2]
```diff
@@ -2,14 +2,19 @@ import { Styleguide } from '../../store/styleguide/styleguide';
...
+ * It then creates pattern folder and pattern instances representing the implementations, and puts
+ * them into the styleguide registry.
  * @see README.md for more details on analyzers and how to write your own.
  */
 export abstract class StyleguideAnalyzer {
 	/**
-	 * Analyzes the pattern implementation directories starting the configured root path, and puts all pattern folders and patterns into the styleguide registry.
-	 * Note: Implementations should call the styleguide's addPattern method, and optionally create new pattern folders based on the styleguide's patternRoot, and also add the pattern to these folders (also addPattern).
+	 * Analyzes the pattern implementation directories starting the configured root path, and puts
+	 * all pattern folders and patterns into the styleguide registry.<br>
```
Comment (LKuechler) [inline src/styleguide/analyzer/styleguide-analyzer.ts]: the `<br>` can be removed right? ^^

**Pull request merged**